docs/integration_testing.rst

```diff
@@ -0,0 +1,157 @@
+Test Components
+---------------
+
+The `conftest.py`_ for integration testing contains the following fixtures, classes and functions utilized by the
+integration test layer. The test classes are used to control state and provide functionality for publishing and
+validation of data in the qTest project under test. The test fixtures wrap the test classes to provide ``pytest`` setup
+and teardown functionality.
+
+Classes
+^^^^^^^
+
+- ``TestCaseInfo``
+    This class models a single JUnitXML style test case produced by the `pytest-rpc`_ plug-in. The class provides
+    properties that map to the data attributes found in a JUnitXML style test case produced by the `pytest-rpc`_
+    plug-in. Methods on the class can be used to validate state on the qTest project under test.
+- ``TestSuiteInfo``
+    This container class models a JUnitXML style test suite produced by the `pytest-rpc`_ plug-in. The class provides
+    properties that map to the data attributes found in a JUnitXML style test suite produced by the `pytest-rpc`_
+    plug-in. The ``add_test_case`` method will add ``TestCaseInfo`` objects to the internal data structure.
+- ``ZigZagRunner``
+    This class does the heavy lifting when it comes to invoking ``ZigZag`` against JUnitXML result files built on the
+    fly. This class wraps the ``TestSuiteInfo`` class and provides pass-through methods for accessing the critical
+    ``TestSuiteInfo`` methods needed for modeling a proper JUnitXML test result file produced by the `pytest-rpc`_
+    plug-in.
+
+Fixtures
+^^^^^^^^
+
+- ``_configure_test_environment``
+    This private fixture is used to prepare the qTest project under test for testing. This fixture is responsible for
+    doing the final clean-up of test cycle artifacts after the `pytest`_ session completes.
+- ``qtest_env_vars``
+    Retrieve a dictionary of required environment variables for running integration tests. If the required environment
+    variables are not present then this fixture will force all tests to fail. (Be aware that the failure message from
+    this fixture is obtuse! Just remember that you need "QTEST_API_TOKEN" and "QTEST_SANDBOX_PROJECT_ID" environment
+    variables populate if this fixture fails)
+- ``_zigzag_runner_factory``
+    This private fixture factory is used as the basis for representing JUnitXML test result files produced by the
+    `pytest-rpc`_ plug-in. This fixture is responsible for doing the final clean-up of test module artifacts after the
+    `pytest`_ session completes.
+
+Helpers
+^^^^^^^
+
+- ``_search_qtest``
+    Search qTest for objects matching a given query. (This helper exists because the swagger_client search API
+    returns a really useless model)
+
+Workflow for Writing a Test Case
+--------------------------------
+
+This section covers the best practices for implementing test cases at the integration layer. This section assumes that
+the developer is already familiar with how to write high quality `pytest`_ test cases.
```

Comment:
Should there be a link to the documentation for writing "high quality `pytest` test cases"?

Reply:
Yeah, if you render the RST it will be a clickable link. The reference is at the bottom of the document.